### PR TITLE
rustdoc: Add support for `missing_unsafe_on_extern` feature

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Follow-up of https://github.com/rust-lang/rust/pull/124482.

Not sure if the `safe` keyword is supposed to be displayed or not though? For now I didn't add it in the generated doc, only `unsafe` as usual.

cc @spastorino 
r? @fmease

### CODE REVIEW

**Comment by spastorino** [comment]
Looks good to me in principle but I think we would want ...

```rust
unsafe extern {
    safe fn foo();
}
```

To actually show in rustdoc that the function is "safe" instead of "". Because "" means unsafe inside extern blocks.

File: tests/rustdoc/unsafe-extern-blocks.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+unsafe extern {
+    // @has 'foo/static.FOO.html'
+    // @has - '//pre[@class="rust item-decl"]' 'pub static FOO: i32'
+    pub safe static FOO: i32;
+    // @has 'foo/static.BAR.html'
+    // @has - '//pre[@class="rust item-decl"]' 'pub unsafe static BAR: i32'
+    pub static BAR: i32;
+
+    // @has 'foo/fn.foo.html'
+    // @has - '//pre[@class="rust item-decl"]' 'pub extern "C" fn foo()'
```
Comment (spastorino) [inline tests/rustdoc/unsafe-extern-blocks.rs:25]: `pub safe extern "C" fn foo()`

File: tests/rustdoc/unsafe-extern-blocks.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    pub safe fn foo();
+    // @has 'foo/fn.bar.html'
+    // @has - '//pre[@class="rust item-decl"]' 'pub unsafe extern "C" fn bar()'
+    pub fn bar();
+}
```
Comment (spastorino) [inline tests/rustdoc/unsafe-extern-blocks.rs:30]: I'd also add `pub unsafe fn baz()` and `pub unsafe static BAZ: i32` to tests.

**Review by spastorino — CHANGES_REQUESTED** [review]

**Comment by GuillaumeGomez** [comment]
> Looks good to me in principle but I think we would want ...
> 
> ```rust
> unsafe extern {
>     safe fn foo();
> }
> ```
> 
> To actually show in rustdoc that the function is "safe" instead of "". Because "" means unsafe inside extern blocks.

I'm a bit shared about this. The important information here I think is whether the function needs an `unsafe` block to be called, not how it's declared. Therefore, the `safe` keyword doesn't provide additional information to the reader in this case. Or did I misunderstand something?

**Comment by spastorino** [comment]
You're right, I forgot that `extern fn foo()` is different than `extern { fn foo() }`.

@bors r+

**Pull request merged**